File: backend/src/race_command_center/models/session.py
from pydantic import BaseModel, Field


class RaceSession(BaseModel):
    session_id: str
    weekend_id: str | None = None
    circuit_id: str
    bike_id: str
    rider_id: str | None = None
    session_type: str
    status: str = "created"
    started_at: str | None = None
    ended_at: str | None = None
    weather: dict = Field(default_factory=dict)
    metadata: dict = Field(default_factory=dict)


class SessionCreate(BaseModel):
    circuit_id: str
    bike_id: str
    rider_id: str | None = None
    session_type: str
    weekend_id: str | None = None
    weather: dict = Field(default_factory=dict)

File: frontend/src/types/session.ts
export type SessionStatus = "created" | "active" | "closed" | "analysed";
export type SessionType = "fp1" | "fp2" | "fp3" | "qualifying" | "race" | "test" | "warmup";

export type RaceSession = {
  session_id: string;
  weekend_id?: string;
  circuit_id: string;
  bike_id: string;
  rider_id?: string;
  session_type: SessionType;
  status: SessionStatus;
  started_at?: string;
  ended_at?: string;
  weather?: Record<string, unknown>;
  metadata?: Record<string, unknown>;
};
